### DISCUSSION TITLE
How to use {{ticks}} in Service JSON

### DISCUSSION THREAD

**Discussion by bchevreau** [cgiesche/streamdeck-homeassistant#271] [Help and Support]
Hi !

Absolute fantastic integration, it worked flawlessly, although I can't for the life of me find how to integrate the ticks variable to manipulate brightness on a SteamDeck Plus dial. Mostly due to my incompetence with JSON :-P

Here is my setup (please correct me if there's anything wrong here), I wanted to add ticks right for +5 brightness, and ticks left (negative) for -5 brightness:

Service JSON:
```
  {
  "brightness": "{% raw %}{{ state_attr('light.living_room_ikea','brightness') + 5 }}{% endraw %}"
  }
```

Domain: light
Service: Turn on
Entity: Ikea Light

Additionally, is there a way to send service JSON for Press-and-hold+Rotate ? So it would handle brightness for the rotate left/right, and color temperature for the presshold left/right rotations?

I would gladly feature this amazing plugin in a video as soon as I manage to figure this out. The coffee is on the way ;-)

**Comment by bchevreau** [comment]
I... sorta hacked my way into it. :-) It's probably shockingly dirty, but it seems to work, here's the JSON for reference:

```
{
"brightness": "{% raw %}{{ state_attr('light.living_room_ikea','brightness') + {% endraw %}{{ ticks }}{% raw %} }}{% endraw %}"
}
```

**Comment by bchevreau** [comment]
Same thing for temperature (obviously set on another dial):

{
"color_temp": "{% raw %}{{ state_attr('light.living_room_ikea','color_temp') + {% endraw %}{{ ticks }}{% raw %} }}{% endraw %}"
}

**Comment by cgiesche** [comment]
Hi!

I have not currently my plus connected, but in theory, there sould be options `brightness_step` and `brightness_step_pct` in the `turn_on` service:

```
{
  "brightness_step": {{ticks}}
}
```

Unfortunately , there is no `_step` option for color_temp, so your hack may be the best option (because it always uses the current color_temp value as a starting point. You can try to use `{{ rotationPercent }}` or `{{ rotationAbsolute }}`. That may make the code more compact, but as a downside, it always starts from 0 when the streamdeck is restarted.

```
{
  "color_temp": {{ rotationAbsolute }}
}

maybe combined with some calculations (only example, but you could manage to define a minimum and a maximum value with some calculation magic):

{
  "color_temp": {{ 9000 / 100 * rotationPercent }}
}
```

**Reply by cgiesche** [reply]
Update: Some time ago, i already did the color-temperature-magic: https://github.com/cgiesche/streamdeck-homeassistant/discussions/209
